token/typed.go

```diff
@@ -0,0 +1,59 @@
+package token
+
+import (
+	"fmt"
+	"sort"
+)
+
+// ArgumentsParser defines a parser for the arguments of a typed token.
+// Parsing stops unrecoverably at the first error. The return value of Err must be checked before using the values returned by precedings calls to GetInt.
+type ArgumentsParser interface {
```

Comment:
Since this is only for typed tokens and in the `token` package, I think this should be renamed to `ArgumentsTyped(Token)Parser`.